### PR TITLE
workload/ycsb: add --insert-start, --insert-count, --record-count options

### LINKED ISSUE #36799: workload: geo-partition support on YCSB
**Is your feature request related to a problem? Please describe.**

opensource YCSB has[ insertstart, insertcount and recordcount](https://github.com/brianfrankcooper/YCSB/wiki/Running-a-Workload-in-Parallel) options.  These options allow YCSB to be used in [parallel geo-partition demos](https://github.com/robert-s-lee/crdbexamples/tree/master/scripts) as well as make the workloads D and E re-entrant.

**Describe the solution you'd like**

add support for the same insertstart, insertcount and recordcount options.
 
**Describe alternatives you've considered**
Currently using opensource YCSB

### PR DESCRIPTION
**Pull request by jeffrey-xiao**
These options allow the YCSB workload to be used in use cases where multiple clients are running workloads in parallel. The initial rows options is subsumed by these options.

I wasn't actually sure what use cases there would be for record count to be not equal to insert start + insert count as reads, scans, and updates might access a key that doesn't exist, but I've added the option to be consistent with the official YCSB implementation.

Example of executing two clients in parallel:

```
$ workload run ycsb --workload e --insert-start 0       --insert-count 500000 --record-count 500000  --max-ops 500000 --pprofport 33333 --drop
$ workload run ycsb --workload d --insert-start 1000000 --insert-count 500000 --record-count 1500000 --max-ops 500000 --pprofport 33334
```

Fixes #36799

cc @robert-s-lee @BramGruneir

### CODE REVIEW

**Review by ajwerner — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38043#-:-LgdU6k7D5HUGcbWa49U:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner, @jeffrey-xiao, and @nvanbenschoten)

___
*[pkg/workload/ycsb/ycsb.go, line 120 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38043#-LgdMn7W-sKi-Pp1Clfw:-LgdMn7XD_h7R7D5bjad:b-rm9ws0) ([raw file](https://github.com/cockroachdb/cockroach/blob/902232582625abf88ee129234916c83d85a286b6/pkg/workload/ycsb/ycsb.go#L120)):*
> ```Go
> 		g.flags.IntVar(&g.insertStart, `insert-start`, 0, `Key to start initial sequential insertions from. (default 0)`)
> 		g.flags.IntVar(&g.insertCount, `insert-count`, 10000, `Number of rows to sequentially insert before beginning workload.`)
> 		g.flags.IntVar(&g.recordCount, `record-count`, 0, `Key is start workload insertions from. Must be >= insert-start + insert-count. (Default: insert-start + insert-count)`)
> ```

s/is/to/
___
*[pkg/workload/ycsb/ycsb.go, line 449 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38043#-LgdTyT19CTC5IJlXNyO:-LgdTyT20t7cfLE68JJE:b-akxn7b) ([raw file](https://github.com/cockroachdb/cockroach/blob/902232582625abf88ee129234916c83d85a286b6/pkg/workload/ycsb/ycsb.go#L449)):*
> ```Go
> 	// TODO(jeffreyxiao): The official YCSB implementation creates a very large
> 	// key space for the zipfian distribution, hashes, mods it by the number of
> 	// expected keys at the end of the workload, and hashes it again to obtain
> ```

I'm not sure I all the way understand this description. Doesn't the mod need to happen at the end to be even close to meaningful?

**Review by jeffrey-xiao — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38043)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner, @jeffrey-xiao, and @nvanbenschoten)

___
*[pkg/workload/ycsb/ycsb.go, line 449 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38043#-LgdTyT19CTC5IJlXNyO:-LgdUchT5Wlz_7DAUx6k:br2b0e5) ([raw file](https://github.com/cockroachdb/cockroach/blob/902232582625abf88ee129234916c83d85a286b6/pkg/workload/ycsb/ycsb.go#L449)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

I'm not sure I all the way understand this description. Doesn't the mod need to happen at the end to be even close to meaningful? 
</blockquote></details>

I've updated the description to be more clear.

In a nutshell

Official YCSB:
```
keyIndex = Hash(zipf.Rand()) % expected_rows
while keyIndex > actual_rows:
  keyIndex = Hash(zipf.Rand()) % expected_rows
key = hash(keyIndex)
```

Our implementation:
```
keyIndex = zipf.Rand() % actual_rows
key = hash(keyIndex)
```

**Review by nvb — APPROVED** [review]
:lgtm:

@robert-s-lee @BramGruneir could you confirm that this allows `workload/ycsb` to be used for geo-partitioned YCSB demos?

> Reviewed 1 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38043#-:-LghXBPjBS43kGgH-JSE:b1oisd3)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner and @jeffrey-xiao)

**Review by BramGruneir — APPROVED** [review]
:lgtm:

This should fix the problem.  Note that when testing for ops/sec, with large distributed system we tend to use uniform distributions instead of zipfian.

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38043#-:-LgnQf2u3FC0BUpW8aIa:bbj0ris)* status: :shipit: complete! 2 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner and @jeffrey-xiao)
